cmd/crc/cmd/start.go

```diff
@@ -96,3 +106,30 @@ func validateStartFlags() error {
 	}
 	return nil
 }
+
+func getPullSecretFileContent() (string, error) {
+	var (
+		pullsecret string
+		err        error
+	)
+	// In case user doesn't provide a file in start command or in config then ask for it.
+	if crcConfig.GetString(config.PullSecretFile.Name) == "" {
+		pullsecret, err = input.PromptUserForSecret("Image pull secret", fmt.Sprintf("Copy it from %s", constants.DefaultPullSecretURL))
+		// This is just to provide a new line after user enter the pull secret.
```

Comment:
> Or does this belong to documentation?

Yes that will go as part of documentation.

> And maybe the logic should be changed to:

I thought of that but it shouldn't be `~/.crc/machines/crc/pull-secret.json` but `~/.crc/cache` since pull secret is not specific to machine, it is not going to change for every start.

Reply:
The advantage of `~/.crc/machines/crc/pull-secret.json` is that `crc delete` would get rid of it in case an invalid secret was given. But you are right that asking for a secret after each `crc delete` may be too much.

Reply:
@cfergeau we do check the validation of pull secret and also it's not only delete but also stop start use case we shouldn't ask user the pull secret if he provided during the start.